primitives/src/balances_map.rs

```diff
@@ -1,5 +1,93 @@
 use std::collections::BTreeMap;
 
-use crate::BigNum;
+use crate::{BigNum, ValidatorId};
+use std::collections::btree_map::{Entry, Iter, Values};
 
-pub type BalancesMap = BTreeMap<String, BigNum>;
+use serde::{ser::SerializeMap, Deserialize, Serialize, Serializer};
+use std::iter::FromIterator;
+use std::ops::Index;
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct BalancesMap(BTreeMap<ValidatorId, BigNum>);
```

Comment:
This is the simplest way of not implementing all the deserialization and serialization, as there is no way atm to specify you want the key of the `BTreeMap` (which already has `Serializer` implemented) to be lowercase or not.
We cannot override the `Serializer` of BTreeMap so we need a newtype for that!